frontend/src/static/js/v_segment.js

```diff
@@ -1,3 +1,21 @@
+window.toggleSegment = function toggleSegment(ele) {
+  // function for toggling unopened segment
+  const targetClass = 'active toggled';
+
+  const parent = ele.parentNode;
+  const classes = parent.className.split(' ');
+  const idxActive = classes.indexOf('active');
+  const idxToggled = classes.indexOf('toggled');
+
+  if (idxActive === -1 && idxToggled === -1) {
+    classes.push(targetClass);
+  } else {
+    classes.splice(idxActive, 2);
```

Comment:
In case this is needed, the way you delete `active` and `toggled` may not be safe, you should delete them separately. It is not guaranteed that the two class are adjacent if the user manually swaps the classes when testing/playing around.